pennylane/qnode.py

```diff
@@ -521,6 +524,16 @@ def construct(self, args, kwargs):
                     " differentiation method"
                 )
 
+        # Apply the deferred measurement principle if the device doesn't
+        # support mid-circuit measurements natively
+        # TODO: change once mid-circuit measurements are not considered as tape
+        # operations

```

Comment:
Yes, or otherwise find a good place for mid-circuit measurements. I think the suggestion to create its own container is nice! One thing to consider here is that mid-circuit measurements cause a complication: we won't be able to simply determine the order in which items are being queued within the circuit unless mid-circuit measurements are queued with operations. Maybe this is okay actually and we just have to make this the official solution, but this TODO more so hints on the fact that wider thinking about how mid-circuit measurements are recorded by tapes is required.